### PR TITLE
fix:  stop request but not calling setStatusCode will trigger an exception In APISIX

### PR DESCRIPTION
**Pull request by tzssangglass**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

___
### Bugfix
- Description
see #55
- How to fix?
Use 500 as the default HTTP Status Code when setStatusCode is not called.
___
### New feature or improvement
- Describe the details and related test reports.

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Comment by tzssangglass** [comment]
@dmsolr pls review it

File: runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java [hunk 1]
```diff
@@ -194,6 +197,14 @@ private int buildStopResp(FlatBufferBuilder builder) {
         Stop.startStop(builder);
         if (!Objects.isNull(statusCode)) {
             Stop.addStatus(builder, statusCode);
+        } else {
+            /**
+             * Avoid APISIX using 0 as the default HTTP Status Code
+             * {@link org.apache.apisix.plugin.runner.HttpResponse#setStatusCode(int statusCode)}
+             * @see https://github.com/apache/apisix-java-plugin-runner/issues/55
+            */
+            Stop.addStatus(builder, 500);
```
Comment (spacewander) [inline runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java:206]: Is 500 a suitable value for the default one?

Comment (tzssangglass) [reply runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java:206]: Not very suitable and can cause some ambiguity, do you have any suggestions？

Comment (tzssangglass) [reply runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java:206]: My options are: 404, 503

Comment (spacewander) [reply runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java:206]: What about 200? Both Go and OpenResty use it.

Comment (tzssangglass) [reply runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java:206]: agree

Comment (tzssangglass) [reply runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java]: I will add this action to APISIX as well.

**Review by dmsolr — APPROVED** [review]
LGTM

File: runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java (hunk 1 above)
Comment (tzssangglass) [reply runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/HttpResponse.java]: update

File: runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersEncoderTest.java [hunk 2]
```diff
@@ -263,6 +263,24 @@ void testEncodeDataGreaterLargeThen256() {
         Assertions.assertEquals(bytes[0], array[1]);
         Assertions.assertEquals(bytes[1], array[2]);
         Assertions.assertEquals(bytes[2], array[3]);
+    }
 
+    @Test
+    @DisplayName("test stop the request without setStatusCode")
+    void testDoFilterWithoutSetStatusCode() {
+        HttpResponse httpResponse = new HttpResponse(0L);
+        // only set header, without setStatusCode, use 500 as default
```
Comment (spacewander) [inline runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersEncoderTest.java:272]:
```suggestion
        // only set header, without setStatusCode, use 200 as default
```

Comment (tzssangglass) [reply runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersEncoderTest.java]: done

**Review by spacewander — APPROVED** [review]

**Pull request merged**